src/base/dev/vga/vgaemu.c

```diff
@@ -1684,8 +1685,29 @@ int vga_emu_pre_init(void)
 
   vga.mem.lfb_base = 0;
   if(config.X_lfb) {
-    dosaddr_t p = alias_mapping_high(MAPPING_VGAEMU,
-				vga.mem.size, VGA_EMU_RW_PROT, vga.mem.base);
+    dosaddr_t p = 0;
+    if (config.cpu_vm_dpmi == CPUVM_KVM) {
+      /* create mapping:
+	 Linear vga.mem.lfb_base=VGAEMU_PHYS_LFB_BASE=0xe0000000 ->
+	 physical 0xe0000000 ->	addr (hugepage aligned) -> vga.mem.base,
+	 so MEM_BASE32(vga.mem.lfb_base) is not a valid pointer. */
+      p = VGAEMU_PHYS_LFB_BASE;
+      void *addr = mmap_mapping(MAPPING_VGAEMU, (void *)-1, vga.mem.size,
+				PROT_NONE);
+      if (addr != MAP_FAILED) {
+	void *addr2 = alias_mapping_ux(MAPPING_VGAEMU, addr, vga.mem.size,
+				       VGA_EMU_RW_PROT, vga.mem.base);
+	if (addr2 != addr)
+	  addr = MAP_FAILED;
+      }
+      if (addr != MAP_FAILED)
+	mmap_kvm(MAPPING_VGAEMU, addr, vga.mem.size, VGA_EMU_RW_PROT, p);
```

Comment:
Why this aliased "addr" needs to be here,
instead of just "vga.mem.base"?
Then this entire code could be removed?